submarineswap.go

```diff
@@ -0,0 +1,96 @@
+package submarineswap
+
+import (
+	"crypto/sha256"
+	"errors"
+	"log"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/input"
+)
+
+const (
+	defaultLockHeight      = 288
+	redeemWitnessInputSize = 1 + 1 + 73 + 1 + 32 + 1 + 100
+	refundWitnessInputSize = 1 + 1 + 73 + 1 + 0 + 1 + 100
+)
+
+var (
+	submarineBucket      = []byte("submarineTransactions")
+	wtxmgrNamespaceKey   = []byte("wtxmgr")
+	waddrmgrNamespaceKey = []byte("waddrmgr")
+)
+
+func genSubmarineSwapScript(swapperPubKey, payerPubKey, hash []byte, lockHeight int64) ([]byte, error) {
+	builder := txscript.NewScriptBuilder()
+
+	builder.AddOp(txscript.OP_HASH160)
+	builder.AddData(input.Ripemd160H(hash))
+	builder.AddOp(txscript.OP_EQUAL) // Leaves 0P1 (true) on the stack if preimage matches
+	builder.AddOp(txscript.OP_IF)
+	builder.AddData(swapperPubKey) // Path taken if preimage matches
+	builder.AddOp(txscript.OP_ELSE)
+	builder.AddInt64(lockHeight)
+	builder.AddOp(txscript.OP_CHECKSEQUENCEVERIFY)
+	builder.AddOp(txscript.OP_DROP)
+	builder.AddData(payerPubKey) // Refund back to payer
+	builder.AddOp(txscript.OP_ENDIF)
+	builder.AddOp(txscript.OP_CHECKSIG)
+
+	return builder.Script()
+}
+
+func NewSubmarineSwap(net *chaincfg.Params, pubKey, hash []byte) (address btcutil.Address, script, swapperPubKey []byte, lockHeight int64, err error) {
+
+	err = pgConnect()
```

Comment:
This connects using a connection pool and should be called once in the main entry point.